pkg/aws/helper.go

```diff
@@ -347,55 +347,7 @@ func (h *AWSHelper) RemoveTargetGroup(tgName string) error {
 	return err
 }
 
-// EnsureUDPTargetGroup ensures that a UDP target group exists with the given port and healt check port
-func (h *AWSHelper) EnsureUDPTargetGroup(vpc, tgName string, port, healthCheckPort int) (string, error) {
-	output, err := h.elbClient.DescribeTargetGroups(&elbv2.DescribeTargetGroupsInput{
-		Names: []*string{aws.String(tgName)},
-	})
-	notFound := false
-	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			if awsErr.Code() == elbv2.ErrCodeTargetGroupNotFoundException {
-				notFound = true
-			}
-		}
-		if !notFound {
-			return "", err
-		}
-	}
-	if !notFound && len(output.TargetGroups) > 0 {
-		tg := output.TargetGroups[0]
-		if aws.Int64Value(tg.Port) == int64(port) {
-			return aws.StringValue(tg.TargetGroupArn), nil
-		}
-		_, err := h.elbClient.DeleteTargetGroup(&elbv2.DeleteTargetGroupInput{
-			TargetGroupArn: tg.TargetGroupArn,
-		})
-		if err != nil {
-			return "", err
-		}
-	}
-	tgResult, err := h.elbClient.CreateTargetGroup(&elbv2.CreateTargetGroupInput{
-		Name:                       aws.String(tgName),
-		Protocol:                   aws.String("UDP"),
-		Port:                       aws.Int64(int64(port)),
-		VpcId:                      aws.String(vpc),
-		TargetType:                 aws.String(elbv2.TargetTypeEnumInstance),
-		HealthCheckProtocol:        aws.String(elbv2.ProtocolEnumTcp),
-		HealthCheckPort:            aws.String(fmt.Sprintf("%d", healthCheckPort)),
-		HealthCheckEnabled:         aws.Bool(true),
-		HealthCheckIntervalSeconds: aws.Int64(10),
-		HealthCheckTimeoutSeconds:  aws.Int64(10),
-		HealthyThresholdCount:      aws.Int64(2),
-		UnhealthyThresholdCount:    aws.Int64(2),
-	})
-	if err != nil {
-		return "", err
-	}
-	return aws.StringValue(tgResult.TargetGroups[0].TargetGroupArn), nil
-}
-
-func (h *AWSHelper) EnsureListener(lbARN, tgARN string, port int, udp bool) error {
+func (h *AWSHelper) EnsureListener(lbARN, tgARN string, port int, tcp bool) error {
```

Comment:
The udp flag is no longer needed since now all listeners are only TCP (which is the default). However, I wouldn't make any further changes to this since we're going to get rid of this code once we switch to a service of type LoadBalancer as @joelddiaz is doing here: https://github.com/openshift-hive/hypershift-installer/pull/8